### DISCUSSION TITLE
I2C api.  SCL and SDA rising edges are sawtooth, not square waveforms;  logic 1 voltage is only 2.5V (not 3.3V)

### DISCUSSION THREAD

**Discussion by timyu1477** [eblot/pyftdi#241] [General]
Hi All, I set up a simple I2C interface with ftdi://ftdi:232h:FT0J4XG6/1   (C232HM-DDHSL-0). Two problems
1. The rising edges of the SCL and SDA are sawtooth, not square waveform.     It seems not be in drive-zero...
2. The logic 1 voltage is pulled down as about 2.5v (probably related with the #1. )
Can someone point out some issue in the configuration? 
Code as following: 

from pyftdi.ftdi import Ftdi 
from pyftdi import i2c
ctrl = i2c.I2cController()   # instantiate FTDI as controller
ctrl.configure('ftdi://ftdi:232h/1') # use standard setup, 3-phase-clocking, drive0, 100KHz etc.

# The DUT address i0x38
poll_status=ctrl.poll(0x38,relax=True)

print(poll_status)

**Comment by eblot** [comment]
What values of the pull-up resistors do you use?

Have you try to execute this script w/o any slave and observed the waveforms?
